frontend/src/components/TradeBox/Forms/LightningPayout.tsx

```diff
@@ -145,44 +146,44 @@ export const LightningPayoutForm = ({
     }
   }, [lightning.lnproxyInvoice, lightning.lnproxyAmount]);
 
-  const lnproxyUrl = function () {
-    const bitcoinNetwork = settings?.network ?? 'mainnet';
-    let internetNetwork: 'Clearnet' | 'I2P' | 'TOR' = 'Clearnet';
+  //filter lnproxies when the network settings are updated
+  let bitcoinNetwork: string = 'mainnet';
+  let internetNetwork: 'Clearnet' | 'I2P' | 'TOR' = 'Clearnet';
+  useEffect(() => {
+    bitcoinNetwork = settings?.network ?? 'mainnet';
     if (settings.host?.includes('.i2p')) {
       internetNetwork = 'I2P';
     } else if (settings.host?.includes('.onion') || window.NativeRobosats != undefined) {
       internetNetwork = 'TOR';
     }
 
-    const url = lnproxies[lightning.lnproxyServer][`${bitcoinNetwork}${internetNetwork}`];
-    if (url != 'undefined') {
-      return url;
-    } else {
-      setBadLnproxyServer(
-        t(`Server not available for {{bitcoinNetwork}} bitcoin over {{internetNetwork}}`, {
-          bitcoinNetwork,
+    filteredProxies = lnproxies
+      .filter((node) => node.relayType == internetNetwork)
+      .filter((node) => node.network == bitcoinNetwork);
+  }, [settings.network]);
+
+  //if "use lnproxy" checkbox is enabled, but there are no matching proxies, enter error state
+  useEffect(() => {
+    setNoMatchingLnProxies('');
+    if (filteredProxies.length === 0) {
+      setNoMatchingLnProxies(
+        t(`No proxies available for {{bitcoinNetwork}} bitcoin over {{internetNetwork}}`, {
+          bitcoinNetwork: settings?.network ?? 'mainnet',
           internetNetwork: t(internetNetwork),
         }),
       );
     }
-  };
-
-  useEffect(() => {
-    setBadLnproxyServer('');
-    lnproxyUrl();
-  }, [lightning.lnproxyServer]);
+  }, [lightning.useLnproxy]);
 
   const fetchLnproxy = function () {
     setLoadingLnproxy(true);
+    let body = {
+      invoice: lightning.lnproxyInvoice,
+      description: '',
+      routing_msat: lightning.lnproxyBudgetSats > 0 ? lightning.lnproxyBudgetSats * 1000 : '',
```

Comment:
Apparently Lnproxy API needs routing budget to be a string:
```
      routing_msat: lightning.lnproxyBudgetSats > 0 ? String(lightning.lnproxyBudgetSats * 1000) : '',
```

In addition, I believe that if the routing budget is 0, you should not include the `routing_msat` param, otherwise Lnproxy's response is "Invalid routing budget"